exporting/exporting_engine.h

```diff
@@ -13,44 +13,62 @@
 
 extern struct config exporting_config;
 
-#define EXPORTER_DATA_SOURCE                 "data source"
-#define EXPORTER_DATA_SOURCE_DEFAULT         "average"
+#define EXPORTER_DATA_SOURCE                    "data source"
+#define EXPORTER_DATA_SOURCE_DEFAULT            "average"
 
-#define EXPORTER_DESTINATION                "destination"
-#define EXPORTER_DESTINATION_DEFAULT        "localhost"
+#define EXPORTER_DESTINATION                    "destination"
+#define EXPORTER_DESTINATION_DEFAULT            "localhost"
 
-#define EXPORTER_UPDATE_EVERY               "update every"
-#define EXPORTER_UPDATE_EVERY_DEFAULT       10
+#define EXPORTER_UPDATE_EVERY                   "update every"
+#define EXPORTER_UPDATE_EVERY_DEFAULT           10
 
-#define EXPORTER_BUF_ONFAIL                 "buffer on failures"
-#define EXPORTER_BUF_ONFAIL_DEFAULT         10
+#define EXPORTER_BUF_ONFAIL                     "buffer on failures"
+#define EXPORTER_BUF_ONFAIL_DEFAULT             10
 
-#define EXPORTER_TIMEOUT_MS                 "timeout ms"
-#define EXPORTER_TIMEOUT_MS_DEFAULT         10000
+#define EXPORTER_TIMEOUT_MS                     "timeout ms"
+#define EXPORTER_TIMEOUT_MS_DEFAULT             10000
 
-#define EXPORTER_SEND_CHART_MATCH           "send charts matching"
-#define EXPORTER_SEND_CHART_MATCH_DEFAULT   "*"
+#define EXPORTER_SEND_CHART_MATCH               "send charts matching"
+#define EXPORTER_SEND_CHART_MATCH_DEFAULT       "*"
 
-#define EXPORTER_SEND_HOST_MATCH            "send hosts matching"
-#define EXPORTER_SEND_HOST_MATCH_DEFAULT    "localhost *"
+#define EXPORTER_SEND_HOST_MATCH                "send hosts matching"
+#define EXPORTER_SEND_HOST_MATCH_DEFAULT        "localhost *"
 
-#define EXPORTER_SEND_NAMES                 "send names instead of ids"
-#define EXPORTER_SEND_NAMES_DEFAULT         CONFIG_BOOLEAN_YES
+#define EXPORTER_SEND_CONFIGURED_LABELS         "send configured labels"
+#define EXPORTER_SEND_CONFIGURED_LABELS_DEFAULT CONFIG_BOOLEAN_YES
+
+#define EXPORTER_SEND_AUTOMATIC_LABELS          "send automatic labels"
+#define EXPORTER_SEND_AUTOMATIC_LABELS_DEFAULT  CONFIG_BOOLEAN_NO
+
+#define EXPORTER_SEND_NAMES                     "send names instead of ids"
+#define EXPORTER_SEND_NAMES_DEFAULT             CONFIG_BOOLEAN_YES
 
 typedef enum exporting_options {
-    EXPORTING_OPTION_NONE              = 0,
+    EXPORTING_OPTION_NONE                   = 0,
+
+    EXPORTING_SOURCE_DATA_AS_COLLECTED      = (1 << 0),
+    EXPORTING_SOURCE_DATA_AVERAGE           = (1 << 1),
+    EXPORTING_SOURCE_DATA_SUM               = (1 << 2),
 
-    EXPORTING_SOURCE_DATA_AS_COLLECTED = (1 << 0),
-    EXPORTING_SOURCE_DATA_AVERAGE      = (1 << 1),
-    EXPORTING_SOURCE_DATA_SUM          = (1 << 2),
+    EXPORTING_OPTION_SEND_CONFIGURED_LABELS = (1 << 3),
+    EXPORTING_OPTION_SEND_AUTOMATIC_LABELS  = (1 << 4),
 
-    EXPORTING_OPTION_SEND_NAMES        = (1 << 16)
+    EXPORTING_OPTION_SEND_NAMES             = (1 << 16)
```

Comment:
Did you jump from 8 (1 << 3) to 32768 ( 1<< 16) because SEND_NAMES will be always last? Or did you do a math on his mind and there is a wrong jump here?

Reply:
I just kept the original values unchanged
https://github.com/netdata/netdata/blob/master/backends/backends.h#L8-L16

Reply:
All right, thanks!